docs/unified-test-documentation/dasharo-security/200-tpm-support.md

```diff
@@ -404,3 +404,374 @@ Example output:
+        22: 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
+        23: 0x0000000000000000000000000000000000000000000000000000000000000000
+    ```
+
+## TPM004.001 PCRALLOCATE Function (Ubuntu 22.04)
+
+**Test description**
+
+This test aims to verify that `PCRALLOCATE` function works properly.
+
+**Test configuration data**
+
+1. `FIRMWARE` = Dasharo
+1. `OPERATING_SYSTEM` = Ubuntu 22.04
+
+**Test setup**
+
+1. Proceed with the
+    [Test cases common documentation](#test-cases-common-documentation) section.
+1. Install `tpm2-tools` using:
+
+    ```bash
+    sudo apt-get install tpm2-tools
+    ```
+
+    [Building from source](https://tpm2-tools.readthedocs.io/en/latest/INSTALL/)
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into the system by using the proper login and password.
+1. Execute command in the terminal:
+
+    ```bash
+    sudo tpm2_pcrallocate
+    ```
+
+1. Note the result
+
+**Expected result**
+
+1. Output from the command shoul look like below:
+
+    ```bash
+    selected-pcrs:
+      - sha1: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23 ]
+      - sha256: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23 ]
+    ```
+
+## TPM005.001 ENCRYPT And DECRYPT Functions (Ubuntu 22.04)
+
+**Test description**
+
+This test aims to verify that ENCRYPT and DECRYPT functions are working
+properly.
+
+**Test configuration data**
+
+1. `FIRMWARE` = Dasharo
+1. `OPERATING_SYSTEM` = Ubuntu 22.04
+
+**Test setup**
+
+1. Proceed with the
+    [Test cases common documentation](#test-cases-common-documentation) section.
+1. Install `tpm2-tools` using:
+
+    ```bash
+    sudo apt-get install tpm2-tools
+    ```
+
+    [Building from source](https://tpm2-tools.readthedocs.io/en/latest/INSTALL/)
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into the system by using the proper login and password.
+1. Execute commands in the terminal:
+
+    ```bash
+    echo "my secret" > secret.dat
+    sudo tpm2_encryptdecrypt -c key.ctx -o secret.enc secret.dat
+    sudo tpm2_encryptdecrypt -d -c key.ctx -o secret.dec secret.enc
+    ```
+
+1. Read `secret.dec` contents:
+
+    ```bash
+    cat secret.dec
+    ```
+
+1. Note the result
+
+**Expected result**
+
+1. Command should output:
+
+    ```bash
+    my secret
+    ```
+
+## TPM006.001 PCREVENT Function (Ubuntu 22.04)
+
+**Test description**
+
+This test aims to verify that PCREVENT function works properly.
+
+**Test configuration data**
+
+1. `FIRMWARE` = Dasharo
+1. `OPERATING_SYSTEM` = Ubuntu 22.04
+
+**Test setup**
+
+1. Proceed with the
+    [Test cases common documentation](#test-cases-common-documentation) section.
+1. Install `tpm2-tools` using:
+
+    ```bash
+    sudo apt-get install tpm2-tools
+    ```
+
+    [Building from source](https://tpm2-tools.readthedocs.io/en/latest/INSTALL/)
```

Comment:
What exactly is this link responsible for?

Reply:
Updated